compiler/rustc_span/src/lib.rs

```diff
@@ -440,6 +440,18 @@ pub struct SpanData {
     pub parent: Option<LocalDefId>,
 }
 
+impl Ord for SpanData {
+    fn cmp(&self, other: &Self) -> Ordering {
+        (self.lo, self.hi, self.ctxt).cmp(&(other.lo, other.hi, other.ctxt))
```

Comment:
Could you explicit destructure so we get a compile error when a field is added to `SpanData`? Can you add a comment on why we ignore `parent`?
`let SpanData { lo: s_lo, hi: s_hi, ctxt: s_ctxt, parent: _ } = self;` and likewise for `other`.